src/controller/CHIPDeviceController.h

```diff
@@ -336,6 +328,22 @@ class DLL_EXPORT DeviceController : public SessionRecoveryDelegate,
 
     OperationalCredentialsDelegate * GetOperationalCredentialsDelegate() { return mOperationalCredentialsDelegate; }
 
+    /**
+     * TEMPORARY - DO NOT USE or if you use please request review on why/how to
+     * officially support such an API.
+     *
+     * This was added to support the 'reuse session' logic in cirque integration
+     * tests however since that is the only client, the correct update is to
+     * use 'ConnectDevice' and wait for connect success/failure inside the CI
+     * logic. The current code does not do that because python was not set up
+     * to wait for timeouts on success/fail, hence this temporary method.
+     *
+     * TODO(andy31415): update cirque test and remove this method.
+     *
+     * Returns success if a session with the given peer does not exist yet.
+     */
+    CHIP_ERROR CloseSession(NodeId nodeId);
```

Comment:
Should this really be construed as API that should never be used? What about the case where a client detects that a CASE session _may_ have gone away on the server, and desires to re-set it up? Wouldn't it perhaps call this before calling `GetConnectedDevice`? Or is there an alternative?

Reply:
It is not used currently by any code except making Cirque pass - there was no equivalent.

My understanding of cirque logic was it tried to do `EstablishConnection` with a immediate `release` when connection succeeds. However this establishes a connection and took time and showed cirque racy. I believe what cirque was doing to release the session may still be a valid `close` but it relies that session already exists (i.e. I cannot do it just by a peer id and did not want to expose the getsession).

So what cirque seemed to try to do:
   - close an active connection (so that new connections start fresh and dont cache anything like peer address)
 
What cirque actually did:
   - Initiate a new CASE connection and because no wait, start resolve while CASE was active (or at least this is how I understood it).
   
This needs python test fixes, but did not add it in this PR, hence this  temporary functions.